src/main/java/net/minecraftforge/client/gui/tabs/GuiTab.java

```diff
@@ -0,0 +1,698 @@
+                }
+                else if (tileEntity.getBlockType() == Blocks.ENDER_CHEST)
+                {
+                    if (tileEntity instanceof TileEntityDropper)
+                    {
+                        mc.displayGuiScreen(new GuiDispenser(mc.player.inventory, (TileEntityDropper) tileEntity));
+                    }
+                }
+            }
+
+        }
+    }.addTo(GuiDispenser.class).setTargetGui(GuiDispenser.class);
+    public static final GuiTab VANILLA_SHULKER_BOX_TAB = new GuiTab(null)
+    {
+        @Override
+        public ItemStack getIconStack()
+        {
+            return new ItemStack(Blocks.PURPLE_SHULKER_BOX);
+        }
+
+        @Override
+        public void onTabClicked(GuiContainer guiContainer)
+        {
+            BlockPos blockPos = mc.objectMouseOver.getBlockPos();
+            TileEntity tileEntity = mc.world.getTileEntity(blockPos);
+            if (tileEntity != null && tileEntity instanceof TileEntityShulkerBox)
+            {
+                mc.displayGuiScreen(new GuiShulkerBox(mc.player.inventory, (TileEntityShulkerBox) tileEntity));
+            }
+        }
+
+        @Override
+        public String getUnlocalizedName()
+        {
+            return "container.shulkerBox";
+        }
+    }.addTo(GuiShulkerBox.class).setTargetGui(GuiShulkerBox.class);
+    public static final GuiTab VANILLA_ENCHANTMENT_TABLE_TAB = new GuiTab(null)
+    {
+        @Override
+        public ItemStack getIconStack()
+        {
+            return new ItemStack(Blocks.ENCHANTING_TABLE);
+        }
+
+        @Override
+        public void onTabClicked(GuiContainer guiContainer)
+        {
+            BlockPos blockPos = mc.objectMouseOver.getBlockPos();
+            TileEntity tileEntity = mc.world.getTileEntity(blockPos);
+            if (tileEntity != null && tileEntity instanceof TileEntityEnchantmentTable)
+            {
+                mc.displayGuiScreen(new GuiEnchantment(mc.player.inventory, mc.world, (TileEntityEnchantmentTable) tileEntity));
+            }
+        }
+
+        @Override
+        public String getUnlocalizedName()
+        {
+            return "container.enchant";
+        }
+    }.addTo(GuiEnchantment.class).setTargetGui(GuiEnchantment.class);
+    public static final GuiTab VANILLA_ANVIL_TAB = new GuiTab(null)
+    {
+        @Override
+        public ItemStack getIconStack()
+        {
+            return new ItemStack(Blocks.ANVIL);
+        }
+
+        @Override
+        public void onTabClicked(GuiContainer guiContainer)
+        {
+            mc.displayGuiScreen(new GuiRepair(mc.player.inventory, mc.world));
+        }
+
+        @Override
+        public String getUnlocalizedName()
+        {
+            return "container.repair";
+        }
+    }.addTo(GuiRepair.class).setTargetGui(GuiRepair.class);
+
+    /**
+     * Internal use for the vanilla default tabs
+     *
+     * @param name The name of the tab
+     */
+    private GuiTab(String name)
```

Comment:
If you move them into another class, you can make this package private so that mods cannot use it.